### PR TITLE
[react-select] Fix loadOptions prop for Async

### PR DESCRIPTION
**Pull request by Methuselah96**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/Methuselah96/react-select/blob/typescript-final/packages/react-select/src/Async.tsx#L31-L40
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/react-select/src/Async.d.ts [hunk 1]
```diff
@@ -5,43 +5,50 @@ import manageState from './stateManager';
...
-     once the promise resolves. */
-  loadOptions: (inputValue: string, callback: ((options: OptionsType<OptionType>) => void)) => Promise<any> | void;
-  /* If cacheOptions is truthy, then the loaded data will be cached. The cache
-     will remain until `cacheOptions` changes value.
-     Default: false. */
-  cacheOptions?: any;
+    /* The default set of options to show before the user starts searching. When
+       set to `true`, the results for loadOptions('') will be autoloaded.
+       Default: false. */
+    defaultOptions?: GroupedOptionsType<OptionType> | OptionsType<OptionType> | boolean;
```
Comment (kylemh) [inline types/react-select/src/Async.d.ts:11]:
```suggestion
    /**
     * The default set of options to show before the user starts searching. When
     * set to `true`, the results for loadOptions('') will be automatically loaded.
     *
     * @default false
     */
    defaultOptions?: GroupedOptionsType<OptionType> | OptionsType<OptionType> | boolean;
```

Before you groan, hear me out! With formatting comments _exactly_ like [this](https://jsdoc.app/), VS Code, Sublime, and JetBrains have an automatic or (often-installed) in-editor integration. Additionally, tools like react-docgen (used by Storybook) can automatically parse the comments and turn them into documentation with their Docs mode.

I'd _love_ if you made this change while here... or at least let's add it to the road map of the TS conversion ticket. I don't making that change. It's so trivial too... Would be a good one to tweet out maybe?!

File: types/react-select/src/Async.d.ts [hunk 2]
```diff
@@ -5,43 +5,50 @@ import manageState from './stateManager';
...
 export const defaultProps: Props<any, boolean>;
 
 export interface State<OptionType extends OptionTypeBase> {
-  defaultOptions?: OptionsType<OptionType>;
-  inputValue: string;
-  isLoading: boolean;
-  loadedInputValue?: string;
-  loadedOptions: OptionsType<OptionType>;
-  passEmptyOptions: boolean;
+    defaultOptions?: OptionsType<OptionType>;
```
Comment (kylemh) [inline types/react-select/src/Async.d.ts:33]: It seems our prettier configs differ 😂 

Would you mind adding a prettier config to the root of this package's typedefs so we dont see format-related diffs in reviews going forward?

**Review by kylemh — APPROVED** [review]

File: types/react-select/src/Async.d.ts (hunk 2 above)
Comment (Methuselah96) [reply types/react-select/src/Async.d.ts:33]: I usually prefer 2 spaces, the 4 spaces is DT specific (see second bullet point [here](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes)).

File: types/react-select/src/Async.d.ts (hunk 1 above)
Comment (Methuselah96) [reply types/react-select/src/Async.d.ts:11]: I'll do it in a follow-up PR. ;)

**Comment by Methuselah96** [comment]
Ready to merge

File: types/react-select/src/Async.d.ts (hunk 2 above)
Comment (kylemh) [reply types/react-select/src/Async.d.ts:33]: Strange 🤔 - perhaps the repo is so large that the root-level prettier config didnt properly pick up in my editor?

File: types/react-select/src/Async.d.ts (hunk 1 above)
Comment (Methuselah96) [reply types/react-select/src/Async.d.ts:11]: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50837

**Pull request merged**